docs/getting_started.md

```diff
@@ -0,0 +1,169 @@
+        'n_rows_to_load': 100,
+        'column_name_mapping': {
+            'image': 'images_raw',
+        },
+        "image_column_names": ["image"],
+        
+    }
+)
+
+my_pipeline.add_op(load_from_hf_hub, dependencies=[])
+```
+
+Two things are happening here:
+1. We created a ComponentOp from the registry. This is a reusable component, we pass it arguments needed to configure it
+
+- **dataset_name**: The name of the dataset on huggingface hub, here we load a [dataset with pokemon images](https://huggingface.co/datasets/huggan/pokemon)
+- **n_rows_to_load**: The number of rows to load from the dataset. This is useful for testing your pipeline on a small scale.
+- **column_name_mapping**: A mapping of the columns in the dataset to the columns in fondant. Here we map the `image` column in the dataset to the `images_raw` subset_column in fondant.
+- **image_column_names**: A list of the original image column names in the dataset. This is used to format the image from the huggingface hub format to a byte string.
+
+
+2. We add our created componentOp to the pipeline using the `add_op` method. This component has no dependencies since it is the first component in our pipeline.
+
+
+Next create a file `load_from_hf_hub/component_spec.yml` with the following content:
+
+```yaml
+name: Load from hub
+description: Component that loads a dataset from the hub
+image: ghcr.io/ml6team/load_from_hf_hub:latest
+
+produces:
+  images: # subset name
+    fields:
+      raw: # field name
+        type: binary # field type
+
+
+args:
+  dataset_name:
+    description: Name of dataset on the hub
+    type: str
+  column_name_mapping:
+    description: Mapping of the consumed hub dataset to fondant column names
+    type: dict
+  image_column_names:
+    description: Optional argument, a list containing the original image column names in case the 
+      dataset on the hub contains them. Used to format the image from HF hub format to a byte string.
+    type: list
+    default: None
+  n_rows_to_load:
+    description: Optional argument that defines the number of rows to load. Useful for testing pipeline runs on a small scale
+    type: int
+    default: None
+```
+
+This is the component spec of the component we have just added to our pipelines, the only thing we have altered is the `produces` section. We have defined what subsets, fields and types this component produces.
+
+Your project should look like this now:
+```
+.
+├── components
+│   └── load_from_hf_hub
+│       └── component_spec.yml
+└── pipeline.py
+```
+
+We now have a fully functional Fondant pipeline, it does not have much functionality yet, but it is a good starting point to build upon. We can already try running this limited example in order to validate our setup.
+
+## Running your pipeline
+
+A Fondant pipeline needs to be compiled before it can be ran. This means translating the user friendly Fondant pipeline definition into something that can be executed by a runner.
```

Comment:
I would leave out the compilation step here and only explain the combined functionality which you get when using `fondant run`.